api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -270,6 +270,29 @@ message HttpConnectionManager {
   // <config_http_conn_man_runtime_represent_ipv4_remote_address_as_ipv4_mapped_ipv6>` for runtime
   // control.
   bool represent_ipv4_remote_address_as_ipv4_mapped_ipv6 = 20;
+
+  // [#not-implemented-hide:]
+  // The configuration for HTTP upgrades.
+  // For each upgrade type desired, an UpgradeConfig must be added.
+  //
+  // .. warning::
+  //
+  //    The current implementation of upgrade headers does not handle
+  //    multi-valued upgrade headers. Support for multi-valued headers may be
+  //    added in future if needed.
+  message UpgradeConfig {
+    // The case-insensitive name of this upgrade, e.g. "websocket".
+    // For each upgrade type present in upgrade_configs, requests with
+    // Upgrade: [upgrade_type]
+    // will be proxied upstream.
+    string upgrade_type = 1;
+    // If present, this represents the filter chain which will be created for
+    // this type of upgrade. If no filters are present, the filter chain for
+    // HTTP connections will be used for this upgrade type.
+    repeated HttpFilter filters = 5;
```

Comment:
Would it be more clear from a config perspective to make this a oneof which can either be a bool (use primary filter chain) or a repeated filter list with in size 1? Just throwing it out there as I think it might be a bit more clear from a config perspective.

Reply:
Hm, initially I thought that'd add clarity but on second thought I think it's sort of weird, because the one-of boolean could really only have one value of true.    Also it's a bit more complicated since I believe oneof can't be repeated so we'd have to have a sub-message for "repeated filter chain".  I'm up for doing the change if you still think it's worth it - let me know!

Reply:
I don't feel strongly about it if you think the current way is better.